aiida/orm/nodes/data/cif.py

```diff
@@ -262,25 +262,14 @@ class CifData(SinglefileData):
     _values = None
     _ase = None
 
-    def __init__(
-        self,
-        ase=None,
-        file=None,
-        filename=None,
-        values=None,
-        source=None,
-        scan_type=None,
-        parse_policy=None,
-        **kwargs
-    ):
+    def __init__(self, ase=None, file=None, filename=None, values=None, scan_type=None, parse_policy=None, **kwargs):
         """Construct a new instance and set the contents to that of the file.
 
         :param file: an absolute filepath or filelike object for CIF.
             Hint: Pass io.BytesIO(b"my string") to construct the SinglefileData directly from a string.
         :param filename: specify filename to use (defaults to name of provided file).
         :param ase: ASE Atoms object to construct the CifData instance from.
         :param values: PyCifRW CifFile object to construct the CifData instance from.
-        :param source:
```

Comment:
Yes, we would lose it, but it shouldn't have been there in the first place. The fact that you can pass `source` is also not in the docstring of any of all the other data plugins. This is something that should simply be documented in the documentation. Docstrings are not the place for this. This is a generic problem with any kind of subclassing system that allows to change signatures.

So I guess that is the real thing missing from this PR. I will look if I can add a section in the docs to document this feature.

Reply:
> Docstrings are not the place for this.

Yeah, I'm still not super sure what docstrings are the place for really. With all the format limitations, the lack of support for expanding docstrings of parent classes instead of just overwriting, etc. I feel like they are quite limited and I am more and more hesitant to dedicate a lot of effort into them.